onnxruntime/core/providers/cpu/tensor/slice.h

```diff
@@ -9,33 +9,33 @@ namespace onnxruntime {
 
 class SliceBase {
  protected:
-  SliceBase(const OpKernelInfo& info) {
-    has_axes_ = info.GetAttrs("axes", axes_).IsOK();
-
-    ORT_ENFORCE(info.GetAttrs("starts", starts_).IsOK(), "Invalid 'starts' attribute value");
-    ORT_ENFORCE(info.GetAttrs("ends", ends_).IsOK(), "Invalid 'ends' attribute value");
-
-    if (has_axes_) {
-      if (axes_.size() > starts_.size())
-        ORT_THROW("'axes' has more entries than the 'starts' attribute holds");
-      if (axes_.size() > ends_.size())
-        ORT_THROW("'axes' has more entries than the 'ends' attribute holds");
-    }
+  SliceBase (const OpKernelInfo& info) {
+    auto has_starts = info.GetAttrs("starts", attr_starts_).IsOK();
+    auto has_ends   = info.GetAttrs("ends",   attr_ends_).IsOK();
+    auto has_axes   = info.GetAttrs("axes",   attr_axes_).IsOK();
+    ORT_ENFORCE(!has_axes || has_starts && has_ends, "Not all required attributes are specified");
```

Comment:
It's wrong.

Please add comment here. It's class is for which ops. And who has which attributes.